src/librustdoc/html/format.rs

```diff
@@ -676,6 +676,36 @@ fn primitive_link(
     Ok(())
 }
 
+crate fn primitive_link_url(
+    prim: clean::PrimitiveType,
+    cx: &Context<'_>,
+    root_path: &str,
+) -> Option<String> {
+    let m = &cx.cache();
+    match m.primitive_locations.get(&prim) {
+        Some(&def_id) if def_id.is_local() => {
+            Some(format!("{}primitive.{}.html", root_path, prim.as_sym(),))
+        }
+        Some(&def_id) => {
+            let path = match m.extern_locations[&def_id.krate] {
+                ExternalLocation::Remote(ref s) => {
+                    format!("{}/std", s.trim_end_matches('/'))
```

Comment:
Hard-coding `/std` definitely seems wrong. This should use `tcx.crate_name(def_id.krate)` instead.